tests/scene/test_code_edit.h

```diff
@@ -2596,6 +2620,51 @@ TEST_CASE("[SceneTree][CodeEdit] folding") {
 		CHECK(code_edit->is_line_folded(0));
 		CHECK_FALSE(code_edit->is_line_folded(1));
 		CHECK(code_edit->get_next_visible_line_offset_from(1, 1) == 4);
+
+		// Non-indented single-line comments.
+		// Indent level 0->1, comment after lines
+		code_edit->set_text("line1\n\tline2\n#test");
+		CHECK(code_edit->can_fold_line(0));
+		CHECK_FALSE(code_edit->can_fold_line(1));
+		code_edit->fold_line(1);
+		CHECK_FALSE(code_edit->is_line_folded(1));
+		code_edit->fold_line(0);
+		CHECK(code_edit->is_line_folded(0));
+		CHECK_FALSE(code_edit->is_line_folded(1));
+		CHECK(code_edit->get_next_visible_line_offset_from(1, 1) == 2);
+
+		// Indent level 0->1, comment between lines
+		code_edit->set_text("line1\n#test\n\tline2\nline3");
+		CHECK(code_edit->can_fold_line(0));
+		CHECK_FALSE(code_edit->can_fold_line(2));
+		code_edit->fold_line(2);
+		CHECK_FALSE(code_edit->is_line_folded(2));
+		code_edit->fold_line(0);
+		CHECK(code_edit->is_line_folded(0));
+		CHECK_FALSE(code_edit->is_line_folded(2));
+		CHECK(code_edit->get_next_visible_line_offset_from(1, 1) == 3);
+
+		// Indent level 1->2, comment after lines
+		code_edit->set_text("\tline1\n\t\tline2\n#test");
+		CHECK(code_edit->can_fold_line(0));
+		CHECK_FALSE(code_edit->can_fold_line(1));
+		code_edit->fold_line(1);
+		CHECK_FALSE(code_edit->is_line_folded(1));
+		code_edit->fold_line(0);
+		CHECK(code_edit->is_line_folded(0));
+		CHECK_FALSE(code_edit->is_line_folded(1));
+		CHECK(code_edit->get_next_visible_line_offset_from(1, 1) == 2);
+
+		// Indent level 1->2, comment between lines
+		code_edit->set_text("\tline1\n#test\n\t\tline2\nline3");
+		CHECK(code_edit->can_fold_line(0));
+		CHECK_FALSE(code_edit->can_fold_line(2));
+		code_edit->fold_line(2);
+		CHECK_FALSE(code_edit->is_line_folded(2));
+		code_edit->fold_line(0);
+		CHECK(code_edit->is_line_folded(0));
+		CHECK_FALSE(code_edit->is_line_folded(2));
+		CHECK(code_edit->get_next_visible_line_offset_from(1, 1) == 3);
```

Comment:
Can group with the other single line comment tests line `~2601`. I'll leave up to you whether this should also test single line strings and or multiline or if that's overkill.

Reply:
Good idea, I agree that would be better grouping. As for the additional testing - it would be good to fully make sure everything works, but I think cloning of these for multiline and strings would significantly clutter up the tests when these use cases are less common. In my opinion it's unlikely that the behaviour would get broken just for strings (or especially just for multiline).